src/Agent.Plugins/RepositoryPlugin.cs

```diff
@@ -214,9 +214,9 @@ public interface ISourceProviderFactory
         ISourceProvider GetSourceProvider(string repositoryType);
     }
 
-    public sealed class SourceProviderFactory : ISourceProviderFactory
+    public class SourceProviderFactory : ISourceProviderFactory
     {
-        public ISourceProvider GetSourceProvider(string repositoryType)
+        public virtual ISourceProvider GetSourceProvider(string repositoryType)
```

Comment:
I haven't gotten very far yet, but the changes to this class seem strange. Why make the only method here virtual when this implements an interface? Couldn't your new implementation simply implement the interface?

Reply:
The reason I did this was because I specifically wanted to be able to call into the base class implementation to allow this code path to be exercised. I could've extended from `ISourceProviderFactory` in `FakeSourceProviderFactory` and created a `SourceProviderFactory` instance in my fake, but this felt a little more natural to me.